src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuck.java

```diff
@@ -446,17 +450,20 @@ private void aggregate(List<Path> files, Path outputFile) throws GoGoDuckExcepti
         }
         else {
             logger.info(String.format("Concatenating %d files into '%s'", files.size(), outputFile));
-            Set<String> fileNames = new HashSet<>();
+            command.addArgument(outputFile.toFile().getAbsolutePath());
+
+            Set<String> fileNames = new LinkedHashSet<>();
+
             for (Path file : files) {
                 String filePath = file.toAbsolutePath().toString();
-                command.add(filePath);
                 fileNames.add(filePath);
             }
-            command.add(outputFile.toFile().getAbsolutePath());
+
+            String stdin = StringUtils.join(fileNames, " ");
 
             // Running ncrcat
             try {
-                execute(command);
+                execute(command, stdin);
             }
             catch (Exception e) {
                 logger.error(String.format("Could not concatenate files: %s into a single file: '%s'", Arrays.toString(fileNames.toArray()), outputFile));
```

Comment:
Yes it will - I'll remove this logging as the GoGoDuckException is logged elsewhere